docs/rfcs/2024-04-01-super-peers-2.md

```diff
@@ -0,0 +1,262 @@
+  point = (h + r)/2 ; as member ID and message hash are uniformly random in 0..1 range, `point` will also be uniformly random in 0..1 range
+  start = (1 - prob) * i / (n - 1) ; `start` for the peer with `i == 0` will be `0` and with `i == n - 1` will be `1 - prob`
+  end = start + prob ; `end` for peer 0 will be `prob` and for peer `n - 1` will be 1
+  ; for all peers the range `start..end` will have width `prob`
+  if (point >= start && point < end) deliver
+  else skip;
+```
+
+This algorithm can be proven to result in target average delivery redundancy.
+
+It also requires all super peers to inform other super peers about:
+- establishing or losing the connection with other members (e.g., when AUTH is received).
+- informing about their decision to stop being super peers in advance.
+- most likely using delivery receipts in communications between super-peers that would include message hashes in RCVD info.
+
+## Authorising administrative changes
+
+As members no longer send messages directly to other members, in addition to the risk of the initial MITM by admin (which is now mitigated by having multiple super-peers) there is a risk of super-peers being compromised at a later stage, and in case of administrative changes (member or group level changes) we have these options:
+
+1. have all members postpone these changes until they are communicated by a sufficient number of super-peers (that would require configuring consensus on the group level).
+2. have admins and owners sign administrative changes with the public key included in their profile during member introduction.
+
+The downside of approach 1 is that administrative actions will be delayed and have to be executed not at the time the message is delivered, but at the time message is confirmed by other super-peers. That is separate and in addition to member consensus for some changes (see below). That also means that groups with one super-peer will have no defence mechanism against super-peer being compromised. That also means that groups with two super-peers will either also have no such defence (in case required consensus level is 1) or will require that both super-peers are available, and no administrative actions will be executed unless both super-peers are available.
+
+The downside of approach 2 is the lack of repudiation, in fact, there is a non-repudiation quality of such administrative changes as role changes, member additions and deletions, and group changes.
+
+Overall, while repudiation of sent messages appears as important, it seems much less important for administrative changes, and signing member and group changes while relying on Merkle DAG for message history integrity appears to be an optimal tradeoff.
+
+To support this functionality the `admin` and `owner` members have to add public keys to their profiles and communicate profile updates to all members (before or after group is switched to super-peers, but probably before is better).
```

Comment:
profile update has also to be communicated to new members when group is switched, does this imply at that point super-peers would be passing these updates, or owners and admins will stay connected to all? (I guess first?)